src/braket/default_simulator/openqasm/interpreter.py

```diff
@@ -470,11 +470,16 @@ def _(self, node: QuantumGateModifier) -> QuantumGateModifier:
 
     @visit.register
     def _(self, node: QuantumMeasurement) -> None:
-        """Doesn't do anything, but may add more functionality in the future"""
+        qubit = node.qubit
+        if isinstance(qubit, IndexedIdentifier):
+            qubit_value = qubit.indices[0][0].value
+            self.context.add_measure(qubit_value)
+        else:  # Identifier
+            self.context.add_measure()
 
     @visit.register
     def _(self, node: QuantumMeasurementStatement) -> None:
-        """Doesn't do anything, but may add more functionality in the future"""
+        self.visit(node.measure)
```

Comment:
this isn't full functionality, as the assignment part of the statement will be ignored. That's alright for now, but I think we should update the comment/docstring to say that the measure is performed, but the assignment is ignored. Maybe even raising a warning

Reply:
Though probably not raising a warning, since the BDK creates IR with measure assignments in the openqasm

Reply:
I added this comment `"""The measure is performed but the assignment is ignored"""`. Do you think it would be worth adding the assignment functionality here as well?

Reply:
No, we can't actually do the assignment in this implementation (there is a feature branch that has assignment used by autoqasm). Here, all the OpenQASM interpreter logic happens as a first pass to create the "circuit", and then that is passed to the simulation. Only after the simulation do we get any measurements, so we would be well past this part of the code